content/en/hosting/4.x/upgrade-troubleshooting.md

```diff
@@ -0,0 +1,151 @@
+---
+title: "Troubleshooting 4.x upgrades"
+linkTitle: "Troubleshooting upgrades"
+weight: 50
+aliases:
+  - 
+description: >
+  What to do when CHT 4.x upgrades don't work as planned
+relatedContent: >
+  hosting/4.x/data-migration
+---
+
+4.0.0 was released in November of 2022 so 4.x is now well into a mature and Medic has learned a number of important lessons on how to unstick 4.x upgrades that get stuck.  Below are some specific tips as well as general practices on upgrading 4.x.
+
+{{% pageinfo %}}
+All tips apply to both [Docker]({{< relref "hosting/4.x/production/docker" >}}) and [Kubernetes]({{< relref "hosting/4.x/production/kubernetes" >}}) based deployments unless otherwise specified.
+
+All upgrades are expected to succeed without issue.  Do not attempt any fixes unless you actively have a problem upgrading.
+{{% /pageinfo %}}
+
+## Considerations
+
+When troubleshooting, consider make sure there are:
+
+* Backups exist and restores have been tested 
+* Extra disk space is availabe (up to 5x!)
+* The upgrade has been tested on a development instance with production data
+
+## A go-to fix: restart
+
+A safe fix for any upgrade getting stuck is to restart all services.  Any views that were being re-indexed will be picked up where they left off without loosing any work.  This should be your first step when trouble shooting a stuck upgrade. 
```

Comment:
should we describe the symptoms of what you mean that an upgrade "gets stuck"?

Reply:
good thinking! since we reference it many times, I moved the definition toward the top of the page in [d6b1228](https://github.com/medic/cht-docs/pull/1709/commits/d6b122820388988630a5a12f030a280ead35e2bd)